### PR TITLE
Rietz/update and delete users

### PR DESCRIPTION
**Pull request by tmrietz**
Wrap up user create/update/delete on admin page
  - Include a db_dump script that includes schema and data
  - Write update and delete functionality on front-end and back-end
  - Handle signatures
  - Make a slight change to the log service that allows null value comparisons

### CODE REVIEW

File: erp-server/src/main/java/com/ttt/erp/service/LogService.java [hunk 1]
```diff
@@ -50,8 +50,8 @@ public PropertyChange(String property, String start, String end) {
         startFields[i].setAccessible(true);
         endFields[i].setAccessible(true);
 
-        String strStart = startFields[i].get(start).toString();
-        String strEnd = startFields[i].get(end).toString();
+        String strStart = startFields[i].get(start) != null ? startFields[i].get(start).toString() : "";
+        String strEnd = startFields[i].get(end) != null ? startFields[i].get(end).toString() : "";
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/LogService.java:54]: fix a log bug where nulls weren't properly handled

File: erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java [hunk 2]
```diff
@@ -30,18 +30,17 @@ public UserAccount userFromPrincipal(Principal principal) {
...
-    * data separately so that it doesn't get
-    *
-    */
+    
     public Optional<UserAccount> createUser(Long userAccountId, UserAccount user) {
         try {
             if (! user.getIsAdmin()) {
                 user.setSignature(this.signatureService.newSignatureForUser(user.getSignature(), user));
+            } else {
+                user.setSignature(null);
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/UserAccountService.java:43]: admin users shouldn't get signatures

File: erp-client/src/app/services/user/user.service.ts [hunk 3]
```diff
@@ -27,4 +27,12 @@ export class UserService {
 	addUser(user: User): Observable<string> {
 		return this.httpClient.post<string>(`${this.baseEndpoint}`, user, {headers: this.jsonHeaders});
```
Comment (taylor-jones) [inline erp-client/src/app/services/user/user.service.ts:28]: I like the idea of using variables here.

File: erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java [hunk 4]
```diff
@@ -60,26 +60,21 @@ public UserAccount getUser() {
...
-    // TODO: potentially this to get users profile?
-//    @GetMapping(value = "/me")
-
-
     // TODO: get actual userId from cookie
     @PutMapping("/{id}")
     public Optional<UserAccount> updateUserAccountById (
-        @CookieValue(value = "userId", defaultValue = "1") String modifiedById,
-        @PathVariable("id") Long awardId,
+        @CookieValue(value = "user") String actingUser,
```
Comment (taylor-jones) [inline erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java:66]: Looks good. Definitely time to get the default user out.

**Review by taylor-jones — APPROVED** [review]
Looks good!

**Pull request merged**